### PR TITLE
add feature to get a single url

### PR DESCRIPTION
**Pull request by PraiseTy**

### CODE REVIEW

File: controllers/main.js [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
+const Url = require('../models/Url');
+const { HTTP_ERRORS } = require('../constant');
 
-const getAllUrls = async(req, res) => {
-    const url = await Url.find()
-    res.json(url)
-}
+const getAllUrls = async (req, res) => {
+  try {
+    const url = await Url.find({});
```
Comment (greatertomi) [inline controllers/main.js:7]:
```suggestion
    const url = await Url.find();
```

File: controllers/main.js [hunk 2]
```diff
@@ -1,35 +1,47 @@
...
-    const url = await Url.findOneAndRemove()
-    res.json(url)
-}
+const getUrl = async (req, res) => {
+  try {
+    const { id: urlId } = req.params;
+    const url = await Url.findOne({ _id: urlId });
+    res.json(url);
+  } catch (error) {
+    res.status(HTTP_ERRORS.NOT_FOUND).json({ error: 'Url cannot be found. Try Again' });
```
Comment (greatertomi) [inline controllers/main.js:20]: The catch block should always be for 500s error

File: controllers/main.js [hunk 3]
```diff
@@ -1,35 +1,47 @@
...
-const getAllUrls = async(req, res) => {
-    const url = await Url.find()
-    res.json(url)
-}
+const getAllUrls = async (req, res) => {
+  try {
+    const url = await Url.find({});
+    res.json(url);
+  } catch (error) {
+    res.status(HTTP_ERRORS.NOT_FOUND).json({ error: 'Something went wrong. Try again' });
```
Comment (greatertomi) [inline controllers/main.js:10]:
```suggestion
    res.status(HTTP_ERRORS.SERVER_ERROR).json({ error: 'Something went wrong. Try again' });
```

File: controllers/main.js [hunk 4]
```diff
@@ -1,35 +1,47 @@
...
-const deleteUrl= async(req, res) => {
-    const { id } = req.params
-    const url = await Url.findOneAndRemove()
-    res.json(url)
-}
+const getUrl = async (req, res) => {
+  try {
+    const { id: urlId } = req.params;
+    const url = await Url.findOne({ _id: urlId });
+    res.json(url);
```
Comment (greatertomi) [inline controllers/main.js:18]:
```suggestion
    const url = await Url.findOne({ _id: urlId });
    
    if (!url) {
      return res.status(HTTP_ERRORS.NOT_FOUND).json({ error: 'Url cannot be found. Try Again' });
    }
    
    res.json(url);
```

**Review by greatertomi — CHANGES_REQUESTED** [review]

File: controllers/main.js (hunk 1 above)
Comment (PraiseTy) [reply controllers/main.js]: done on 51b5b13

File: controllers/main.js (hunk 2 above)
Comment (PraiseTy) [reply controllers/main.js]: done on 2d31142

File: controllers/main.js (hunk 4 above)
Comment (PraiseTy) [reply controllers/main.js]: done on 2d31142

File: controllers/main.js (hunk 3 above)
Comment (PraiseTy) [reply controllers/main.js]: done on  2d31142

**Review by greatertomi — APPROVED** [review]

**Pull request merged**